mathesar_ui/src/pages/schemas/schema-row/SchemaRow.svelte

```diff
@@ -0,0 +1,57 @@
+<script lang="ts">
+  import {
+    faLock,
+    faProjectDiagram,
+    faTrash,
+    faPencilAlt,
+  } from '@fortawesome/free-solid-svg-icons';
+  import { Icon, Button } from '@mathesar-components';
+  import type { Schema } from '@mathesar/utils/preloadData';
+
+  // Props
+  export let schema: Schema;
+
+  $: isDefault = schema.name === 'public';
+  $: isLocked = schema.name === 'public';
```

Comment:
Since both the conditions are the same, it would suffice to use only one computed property here.

Currently, since we only are limiting options to the public schema, it's okay to hardcode it on the client. @kgodey It would be better if we can get these from the API.

Reply:
When these are supplied by the API, I assume we could have the functionality to lock non-default schemas as well? In that case it'll be easier if we keep the locking and default as two separate attributes of a schema.

Reply:
Yes, in that case, the schema object will have the attributes, and we wouldn't require computed properties.

Reply:
I guess it's okay to keep it for now. I see no issues in having an additional property.